File: csharp/pangram/Pangram.cs
using System.Collections.Generic;

public class Pangram
{
    private const string Alphabet = "abcdefghijklmnopqrstuvwxyz";

    public static bool IsPangram(string input)
    {
        return new HashSet<char>(input.ToLower()).IsSupersetOf(Alphabet);
    }
}

File: c/pangram/src/pangram.c
#include <stdio.h>
#include "pangram.h"

bool is_pangram(const char *const sentence)
{
    if (sentence == NULL)
    {
        return false;
    }

    int repetitions[26] = { 0 };

    for (size_t i = 0; sentence[i] != '\0'; i++)
    {
        if (isalpha(sentence[i]))
        {
            repetitions[tolower(sentence[i]) - 'a']++;
        }
    }

    for (size_t i = 0; i < 26; i++)
    {
        if (repetitions[i] == 0)
        {
            return false;
        }
    }

    return true;
}
